junit-platform-surefire-provider/src/test/java/org/junit/platform/surefire/provider/RunListenerAdapterTests.java

```diff
@@ -92,13 +95,17 @@ void notifiedWhenMethodExecutionSkipped() throws Exception {
 	@Test
 	void notifiedWithCorrectNamesWhenClassExecutionSkipped() throws Exception {
 		ArgumentCaptor<ReportEntry> entryCaptor = ArgumentCaptor.forClass(ReportEntry.class);
+		TestPlan testPlan = TestPlan.from(Collections.singletonList(new EngineDescriptor(newId(), "Luke's Plan")));
+		adapter.testPlanExecutionStarted(testPlan);
+
+		TestIdentifier classIdentifier = newClassIdentifierOnTestPlan(testPlan);
 
-		adapter.executionSkipped(newClassIdentifier(), "test");
+		adapter.executionSkipped(classIdentifier, "test");
 		verify(listener).testSkipped(entryCaptor.capture());
 
 		ReportEntry entry = entryCaptor.getValue();
 		assertTrue(MyTestClass.class.getTypeName().contains(entry.getName()));
-		assertEquals(MyTestClass.class.getName(), entry.getSourceName());
+		assertEquals("<unrooted>", entry.getSourceName());
```

Comment:
This test's behaviour is now wrong, the result is "<unrooted>", because the class identifier has no parent in the test fixture. However for the right name to appear here the class' parent would need to be itself.

@marcphilipp how do you propose we fix this, or was the test wrong in the first place?